### PR TITLE
GCS_MAVLink: fill mission type in in return MISSION_ITEM packets 

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
@OXINARF I fixed the commit message - but also found there were several other problems with the MISSION_ITEM layer.  I've pushed up patches to this PR for that.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1533,6 +1533,16 @@ void GCS_MAVLINK::send_rc_channels() const
...
 }
 
+bool GCS_MAVLINK::sending_mavlink1() const
+{
+    const mavlink_status_t *status = mavlink_get_channel_status(chan);
+    if (status == nullptr) {
+        // should not happen
+        return true;
+    }
+    return status->flags & MAVLINK_STATUS_FLAG_OUT_MAVLINK1;
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: needs to have a != 0 to be a bool

File: libraries/GCS_MAVLink/MissionItemProtocol.cpp [hunk 2]
```diff
@@ -36,6 +36,13 @@ void MissionItemProtocol::handle_mission_count(
     const mavlink_mission_count_t &packet,
     const mavlink_message_t &msg)
 {
+    // need mavlink2 to do mission types other than mission:
```
Comment (WickedShell) [inline libraries/GCS_MAVLink/MissionItemProtocol.cpp]: Maybe extract this as a helper/macro? Seems to be here verbatim 3 times.

**Comment by tridge** [comment]
needs testing for mission download/upload with --mav10

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Fixed.

File: libraries/GCS_MAVLink/MissionItemProtocol.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/MissionItemProtocol.cpp]: Made a helper for it.

**Comment by peterbarker** [comment]
`--mav10` works just fine

**Pull request merged**